test2code/src/test/kotlin/com/epam/drill/test2code/coverage/InMemoryRetentionQueueTest.kt

```diff
@@ -0,0 +1,163 @@
+        val byteCount = 6
+        val innerArraySize = 2
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+        val sizeLimit = BigInteger.valueOf(10)
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = sizeLimit)
+
+        queue.addAll(bytes)
+
+        val actualSizeOfQueue = queue.flush().sumOf { it.size }
+        Assertions.assertEquals(sizeLimit.toInt(), actualSizeOfQueue)
+    }
+
+    @Test
+    fun `adding elements with size equal to limit should result in queue size exactly at limit`() {
+        val byteCount = 1
+        val innerArraySize = 10
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+        val sizeLimit = BigInteger.valueOf(10)
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = sizeLimit)
+
+        queue.addAll(bytes)
+
+        val actualSizeOfQueue = queue.flush().sumOf { it.size }
+        Assertions.assertEquals(sizeLimit.toInt(), actualSizeOfQueue)
+    }
+
+    @Test
+    fun `adding single element larger than limit should result in an empty queue`() {
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = BigInteger.valueOf(10))
+        val byteCount = 1
+        val innerArraySize = 12
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+
+        queue.addAll(bytes)
+
+        Assertions.assertTrue(queue.flush().count() == 0)
+    }
+
+    @Test
+    fun `flush should return the same elements as added to the queue`() {
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = BigInteger.valueOf(10))
+        val byteCount = 1
+        val innerArraySize = 5
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+
+        queue.addAll(bytes)
+        val actualBytes = queue.flush()
+
+        Assertions.assertEquals(bytes.toList(), actualBytes.toList())
+    }
+
+    @Test
+    fun `queue after flush must be empty`() {
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = BigInteger.valueOf(10))
+        val byteCount = 1
+        val innerArraySize = 5
+        val bytes = Array(byteCount) { ByteArray(innerArraySize) { 1 } }.asSequence()
+
+        queue.addAll(bytes)
+
+        val bytes1 = queue.flush().toList()
+        val bytes2 = queue.flush().toList()
+
+        Assertions.assertTrue(bytes2.isEmpty())
+    }
+
+    @Test
+    fun `queue should only fit elements with size less than or equal to total limit`() {
+        val queue = InMemoryRetentionQueue(totalSizeByteLimit = BigInteger.valueOf(10))
+        val expectedSizeOfQueue = 4
+        val bytes = sequenceOf(
+            ByteArray(12) { 1 },
+            ByteArray(2) { 1 },
+            ByteArray(2) { 1 }
+        )
+
+        queue.addAll(bytes)
+        val actualSizeOfQueue = queue.flush().sumOf { it.size }
+
+        Assertions.assertEquals(expectedSizeOfQueue, actualSizeOfQueue)
```

Comment:
do not introduce `expectedSizeOfQueue` just to hold the expected value

Reply:
done